packages/cli/src/handlers/oauthLogin.ts

```diff
@@ -0,0 +1,374 @@
+import {
+    AuthorizationError,
+    CreatePersonalAccessToken,
+    formatDate,
+    oauthPageStyles,
+} from '@lightdash/common';
+import { exec } from 'child_process';
+import * as crypto from 'crypto';
+import * as http from 'http';
+import { createServer } from 'net';
+import fetch from 'node-fetch';
+import { URL } from 'url';
+import { promisify } from 'util';
+import GlobalState from '../globalState';
+import * as styles from '../styles';
+
+// Shared CSS styles for OAuth response pages - matching Lightdash login design
+
+const execAsync = promisify(exec);
+
+// PKCE helper functions
+const generateCodeVerifier = (): string =>
+    crypto.randomBytes(32).toString('base64url');
+
+const generateCodeChallenge = (verifier: string): string =>
+    crypto.createHash('sha256').update(verifier).digest('base64url');
+
+const generatePersonalAccessToken = async (
+    accessToken: string,
+    url: string,
+): Promise<string> => {
+    const patUrl = new URL(`/api/v1/user/me/personal-access-tokens`, url).href;
+    const now = new Date();
+    const createToken: CreatePersonalAccessToken = {
+        expiresAt: new Date(now.setDate(now.getDate() + 30)), // 30 days from now
+        description: `Generated by OAuth on ${formatDate(now)}`,
+        autoGenerated: true,
+    };
+    const body = JSON.stringify(createToken);
+    const patResponse = await fetch(patUrl, {
+        method: 'POST',
+        body,
+        headers: {
+            'Content-Type': 'application/json',
+            Authorization: `Bearer ${accessToken}`,
+        },
+    });
+    if (!patResponse.ok) {
+        throw new AuthorizationError(
+            `Failed to create PAT: ${patResponse.status} ${patResponse.statusText}`,
+        );
+    }
+    const patResponseBody = await patResponse.json();
+    const { token } = patResponseBody.results;
+    return token;
+};
+// Helper function to find an available port
+const findAvailablePort = async (
```

Comment:
i'm pretty sure the standard node server starts on a random high number port if it's not specified! we shouldn't need this custom port finding